### PR TITLE
Fix the TagOlderThan tests and add some further tests

### PR DESCRIPTION
**Pull request by peternewman**
After #3964 !

### CODE REVIEW

File: app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/TagOlderThanTest.kt [hunk 1]
```diff
@@ -69,11 +69,23 @@ class TagOlderThanTest {
...
+        assertFalse(c.matches(mapOf("foo" to "bar"), newDate))
+    }
+
+    @Test fun `matches element with no tags which is old`() {
+        assertTrue(c.matches(mapOf(), oldDate))
+    }
+
+    @Test fun `does not match element with no tags which is new`() {
+        assertFalse(c.matches(mapOf(), newDate))
+    }
```
Comment (FloEdelmann) [inline app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/TagOlderThanTest.kt:90]: Better test names and grouping:

```suggestion
    @Test fun `matches any old element without relevant tags`() {
        assertTrue(c.matches(mapOf("foo" to "bar"), oldDate))
        assertTrue(c.matches(mapOf(), oldDate))
    }

    @Test fun `does not match any new element without relevant tags`() {
        assertFalse(c.matches(mapOf("foo" to "bar"), newDate))
        assertFalse(c.matches(mapOf(), newDate))
    }
```

Comment (peternewman) [reply app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/TagOlderThanTest.kt]: Makes sense. I think in theory the empty maps should never exist...

File: app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/TagOlderThanTest.kt [hunk 2]
```diff
@@ -69,11 +69,17 @@ class TagOlderThanTest {
...
 
-    @Test fun `does not match old element with tag only as a prefix`() {
-        assertFalse(c.matches(mapOf("opening_hours:signed" to "tag"), oldDate))
-    }
-
     @Test fun `does not match new element with tag only as a prefix`() {
         assertFalse(c.matches(mapOf("opening_hours:signed" to "tag"), newDate))
     }
+
+    @Test fun `matches any old element without relevant tags`() {
```
Comment (peternewman) [inline app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/TagOlderThanTest.kt:76]: @westnordost , just to check that this is intended behaviour? It caught me by surprise a bit (especially given the function name), although doesn't impact on the current usages as they are also using a separate element filter.

Comment (westnordost) [reply app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/TagOlderThanTest.kt:76]: Hmm, I guess this behavior would be unexpected.

Comment (peternewman) [reply app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/TagOlderThanTest.kt:76]: I think if its fixed #3946 will need some alternative logic for when recycling/cycleway don't have a check date already.

I guess doing `TagOlderThan || ElementOlderThan` would make sense, or `AnyTagOlderThan("recycling:*")`

Comment (westnordost) [reply app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/TagOlderThanTest.kt:76]: Hmm... right. `check_date:cycleway` could be set but that doesn't mean that `cycleway` is set

Comment (westnordost) [reply app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/TagOlderThanTest.kt:76]: Besides, the parser does this:

https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/data/elementfilter/ElementFiltersParser.kt#L243

so when written in the element filter expression, it **does** check for the tag being set. Hence, no documentation update in 

https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/data/elementfilter/ElementFilterExpression.kt#L25-L61

is necessary.

**Pull request merged**